### ISSUE TITLE
get configuration from the environment

### ISSUE THREAD

**Issue by japaric** [cross-rs/cross#147]
Today Cross can be configured via the Cross.toml file. However, we would also like to be able to specify this configuration via environment variables.

### What to do

- Each configuration entry allowed in Cross.toml should be overrided by a corresponding environment variable, if set. If the configuration entry is named `build.env.passthrough` then its environment variable counterpart will be named `CROSS_BUILD_ENV_PASSTHROUGH`. If the configuration entry takes a list of strings as value then the env variable value will be parsed as a list of *space* separated strings.

### References

- The Cross.toml parsing logic is in `src/main.rs`, in the `impl Toml` block.
- Cargo has a similar behavior. You'll probably be able to borrow some logic from there.

Required for japaric/trust#73

**Comment by olitha** [comment]
Hi Japaric,

I would like to set images based on target devices that have the same architecture but should be built in different containers. This could be done with setting an env variable to select the build image.

So I am interested in this feature and I would like to give it a try to implement it.

I had a look in the cargo source and it looks for me as there the env vars are read when they are used. My idea instead is to use a `struct Config` that holds both the env vars and the `Toml`. It acts then as a facade and checks for vars first, then the toml.
Signatures of functions then change because I pass the Config.
The only limit I see with this, is that no env vars could be set during the build process that could change the build in later steps, like in a build.rs. But this is anyway not relevant for cross probably.

Do you see any downside with this?

**Comment by olitha** [comment]
Right now for the xargo flag the `[build]]xargo` value overrides the values in target. So there is no way to specify a default.

I wonder if it would be more intuitive to use the most special value, meaning:
```
[build]
xargo = true
[target.aarch64-unknown-linux-gnu]
xargo = false
```
instead of currently using `xargo=true` always, it would use `xargo=true` for all targets that are not further specified, but `xargo=false` for target aarch64-unknown-linux-gnu

**Comment by Alexhuszagh** [comment]
Closed in #525.
